### ISSUE TITLE
Youtube video available?

### ISSUE THREAD

**Issue by johanf85** [crgarcia12/electronics-homeassistant-lightscontroll#2]
Hi, 

Very interesting project. I was looking into a similar solution for using regular light switches in combination with switching from within Home Assistant. I made a forum post in which I also mentioned your repo. See here: https://community.home-assistant.io/t/converting-light-switches/90337/22?u=johanf

Do you maybe have a Youtube video somewhere in which you demonstrate the PCB?

**Comment by crgarcia12** [comment]
Hi @johanf85, 
Thanks for the message!

I am travelling this week, so I don't have anything with me. Next week, I can upload some pictures to show the real PCB.
I never thought on making a video, but since this is the first "real" PCB I design, it took me many many months, pitfalls trial-and-error and learnings (like clearance, coating, digital signals, etc) to bring it to the current state.

Regarding your post, I like it very much! 
I also thought on changing my main detections by a CT clamp. I actually bought bunch of them and tested. Doing that has the advantage that it can inform current consumption. One problem that I have, is that it makes the box a bit more messy with wires dangling around, and makes the installation more expensive... but happy to take a look at this approach if you think it would add value. What do you think?

Regarding the Shelly PRO: 
I have other shellys and, in general, I like the product very very much. 
In the other hand, I have had some problems with them:
- They get quite hot, which I assume it reduces their lifetime, and also does not make me feel confortable to have many hot devices in the mains board. I took a look at some published support cases, and I also saw many of them burning (not the PRO) or sparking. Understandable actually, I would not be able to put in such small box all what Shellys can do.
- If the Shelly burns, my house stop working. This is not aceptable for me, I want a "side-car" setup. Also, I am not sure if I will be able to buy a Shelly replacement in 5/10 years. In this project I uploaded the 3 order files are needed to get a fully assembled PCBs from PCBWay or partially PCB from JLCPCB. 
In the worse case, if there are component you cannot find any longer, then you can simply remove the device and the house keeps working as it used to do before.

I also analyze other solutions, like DigitalStrom or KNX. They look very good, and also have their benefits and cons

Cheers!

**Comment by johanf85** [comment]
Hi Carlos, 

Yes that would be nice to see! Interesting project you are doing. I am nowhere near the skills of designing a PCB board for this. 

And it's an interesting idea to supply the files so others can get their PCB assembled in the future, so that users are not dependent on the continuing of a product by a manufacturer.  

Regarding the non-contact voltage detectors, I think the interesting part about that is that you don't have to connect the high voltage to a PCB. But you are right when it comes to switching a lot of lights, that it complicates the setup. I guess there are just multiple ways to go. 

Maybe something to consider is the solution with a double switching relay like the one used here: https://maker.pro/raspberry-pi/projects/an-automated-light-switch-system-for-smarthomes-that-anyone-can-use 

His relay, the finder 26.2, switches two circuits at once on every impulse. One circuit for the load (light) and one circuit with a input to a micro controller. 

In that way there is no need to run a High voltage wire to a circuit of the microcontroller unit. However this design doesn't give information on whether the lamp is actually on, only whether the relay is in the on-position. But that is distinction I am willing to be okay with. 

Yeah I heard about Digitalstrom too, but to me it seemed expensive and there is also a dependancy of a manufacturer. 

Good luck on this project and your home. 

Johan

**Comment by crgarcia12** [comment]
Hi @johanf85, sorry for the delay, hard to find some time to work on this these weeks :S
 
The 3 files needed to get this PCB fully manufactured and assembled are in this repo. I added a GIF showing how you can order it on JLCPCB. If somebody cannot solder at all, there are services like PCBWay that can fully produce the board.
I also added some more pictures of the real PCB in the main readme.

I did a quick test on the non-contact voltage detectors. I like the idea very much, and it should be very easy to adapt my PCB to that. Once I have a bit more time I will try to do it, also because it will allow to measure consumption on the devices as well.

I looked at the double switching relay. For what I want I am using this device, it sounds too dangerous to mix mains and low voltage wires. Any mistake and something might go very wrong (I could write an entire book on the numerous stupid mistakes I have done and the tons-of devices I have blown :S )

That is why I added protections against mistakes in every input/output in the PCB. For example: 
- Every sensing port has 330k resistance, an 80mA fuse and an optocoupler for isolation. In other words, the mains voltage never reaches the low-voltage sector, and I am pretty confident that it can hold thousands of volts without catching fire or exploding, even if the wires are connected wrongly.
- Outbound relays have 80 or 800mA slow blown fuses. The wet input has a 250mA fast-blowing fuse and a varistor. Pretty safe as well I think. If you connect the relay to the wrong device or plug the wrong wire, you will break the fuse, but probably nothing else.
- The low voltage Power Supply (200V->5v) is a well known brand with 4kV insulation resistance. It does not really need any pretection, but I anyways added a second fast fuse before it. I think this is quite a lot of protection for the microcontroller. I also checked with the oscilloscope and the noise is very low, because of all the capacitors around.

I did some security testing already, for example, I used a Insulation tester passing 1000V. Nothing popped. 
I will do it again and take some pictures or a video and post it here. Maybe I can even make a worse case connection as well, to see how does the board reacts if I short-circuit everything :)

Another detail is that I am also coating the PCB with a 90kV/mm2 to avoid any arching. I think this is completely unnecessary because the board is quite protected already, but since it also cost nothing, why not?

In summary, my main goal is to try to not to let any input/output unprotected and consider all worse cases, even making wrong connections or pushing the device to the limits. The idea is that in the worse case you burn a varistor or a fuse, but no people should be in danger and nothing catches fire

I sent some PCBs for manufacture now with the latest design. Not sure where you live, but I could post you an empty one if  you want.

I am still thinking on how to make some video explaining all the design details and showing the assembly and the device working. I could also record the testing of all the protections and common wiring mistakes. The problem is that I don't have any cameras or experience building such video, so it will take a bit of time until I can do it
